api/alert_rules.go

```diff
@@ -0,0 +1,177 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// AlertRulesService is the service that interacts with
+// the AlertRules schema from the Lacework APIv2 Server
+type AlertRulesService struct {
+	client *Client
+}
+
+type alertRuleSeverity int
+
+type AlertRuleSeverities []alertRuleSeverity
+
+const alertRuleEventType = "Event"
+
+func (sevs AlertRuleSeverities) toInt() []int {
+	var res []int
+	for _, i := range sevs {
+		res = append(res, int(i))
+	}
+	return res
+}
+
+const (
+	AlertRuleSeverityCritical alertRuleSeverity = 1
+	AlertRuleSeverityHigh     alertRuleSeverity = 2
+	AlertRuleSeverityMedium   alertRuleSeverity = 3
+	AlertRuleSeverityLow      alertRuleSeverity = 4
+	AlertRuleSeverityInfo     alertRuleSeverity = 5
+)
+
+// NewAlertRule returns an instance of the AlertRuleData struct
+//
+// Basic usage: Initialize a new AlertRuleData struct, then
+//              use the new instance to do CRUD operations
+//
+//   client, err := api.NewClient("account")
+//   if err != nil {
+//     return err
+//   }
+//
+//   alertRule := api.NewAlertRule(
+//		"Foo",
+//		api.AlertRule{
+//		Description: "My Alert Rule"
+//		Severities: api.AlertRuleSeverities{api.AlertRuleSeverityHigh,
+//		Channels: []string{"TECHALLY_000000000000AAAAAAAAAAAAAAAAAAAA"},
+//		ResourceGroups: []string{"TECHALLY_111111111111AAAAAAAAAAAAAAAAAAAA"}
+//       },
+//     },
+//   )
+//
+//   client.V2.AlertRules.Create(alertRule)
+//
+func NewAlertRule(name string, rule AlertRule) AlertRuleData {
```

Comment:
I feel that the `NewAlertRule()` function should return the struct `AlertRule` so that means that we can rename the current `AlertRule` struct which acts like parameters of the function as:
* `AlertRuleConfig`
* `AlertRuleParams`
* `AlertRuleParameters`
* `AlertRuleDetails`
* `AlertRuleInfo`

NOTE: I like the first two. cc @vatasha thoughts? 

```suggestion
func NewAlertRule(name string, rule AlertRuleConfig) AlertRule {
```

Reply:
I like it! `AlertRuleConfig` would be my preference